cmd/soci/commands/rebuild_db.go

```diff
@@ -0,0 +1,155 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package commands
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/awslabs/soci-snapshotter/fs/config"
+	"github.com/awslabs/soci-snapshotter/soci"
+	"github.com/containerd/containerd/cmd/ctr/commands"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	"github.com/opencontainers/go-digest"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/urfave/cli"
+	"oras.land/oras-go/v2/content/oci"
+)
+
+var RebuildDBCommand = cli.Command{
+	Name:  "rebuild-db",
+	Usage: "rebuild the artifacts database",
+	Action: func(cliContext *cli.Context) error {
+		client, ctx, cancel, err := commands.NewClient(cliContext)
+		if err != nil {
+			return err
+		}
+		defer cancel()
+
+		artifactsDb, err := soci.NewDB(soci.ArtifactsDbPath())
+		if err != nil {
+			return err
+		}
+		blobStore, err := oci.New(config.SociContentStorePath)
+		if err != nil {
+			return err
+		}
+		blobStorePath := filepath.Join(config.SociContentStorePath, "blobs")
+		// check if artifact db contains all artifacts in content store, if not add them
+		err = filepath.WalkDir(blobStorePath, func(path string, d fs.DirEntry, err error) error {
+			if !d.IsDir() {
+				info, err := d.Info()
+				if err != nil {
+					return err
+				}
+				f, err := os.Open(path)
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+				var sociIndex soci.Index
+				err = soci.DecodeIndex(f, &sociIndex)
+				// skip: file is either a ztoc or empty config
+				if err != nil || info.Size() < 10 {
```

Comment:
should the `info.Size() < 10` check more close to where `info` is created (line 63)